### DISCUSSION TITLE
How to use custom syntax highlighting?

### DISCUSSION THREAD

**Discussion by NotroDev** [AvaloniaUI/AvaloniaEdit#367] [Q&A]
I'm trying to use AvaloniaEdit, and it works fine, but I can't get my custom syntax highlighting to work.

What functions do I need to use? Which files?

I have all the syntax files, but I don't know which I need.

I'm really sorry for this very basic question, but I really tried a whole day, and it didn't work.

I know that there is this instruction in the README, but I don't understand what exactly I should do.

**Comment by NotroDev** [comment]
Okay, it works with the simple, default Highlighting Loader from original AvalonEdit. If I find out how to use TextMate, I'll probably rewrite it, tho.

**Reply by MihaMarkic** [reply]
Try using a `DocumentColorizingTransformer` subclass, override `ColorizeLine` method which can call `ChangeLinePart` method. Add this class to `TextArea.TextView.LineTransformers`.

**Reply by NotroDev** [reply]
Oh, thanks for the response!
I've been using the default syntax highlighting system and everything was fine. Recently I started rewriting it to Textmate and I managed to understand how it works.
